test/e2e/configs.go

```diff
@@ -205,6 +205,7 @@ func createRulesYAML(
 const recordingRuleYamlTpl = `
 groups:
   - name: example
+    interval: 30s
```

Comment:
Had to update the tests a bit with rule group `interval` and alert `for` and `annotations` fields. Currently, as it's defined in [spec.yaml](https://github.com/observatorium/api/blob/main/rules/spec.yaml#L83), all these fields are `required` which means that (un)marshaling them will lead to `interval: ""` or `annotations: {}` if they aren't provided by the tenant.

Is there a way to handle this better or allow tenant to not have to provide these fields?

Reply:
Wait, I'm confused here, so the API is accepting a rule file even if those empty parameters are required? That does not sound right.

Reply:
We clarified this F2F with @saswatamcode as of now this is indeed how it works in API, but it is outside the scope of this PR, resolving.